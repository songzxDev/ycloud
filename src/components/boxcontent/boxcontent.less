.y-box-content{
  background: #FFFFFF;
  margin-bottom: 10px;
  padding-bottom: 20px;
  border: solid 1px @border-color;
  padding: 15px 20px 20px 20px;
  .y-box-content-title{
    border-left: solid 4px @danger-color;
    margin-bottom: 18px;
    font-size: 16px;
    padding-left: 10px;
  }
}
