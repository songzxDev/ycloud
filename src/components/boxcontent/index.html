<div class="y-box-content">
  <!-- ko if: title -->
  <div class="y-box-content-title" data-bind="text:title"></div>
  <!-- /ko -->
  <!-- ko template: { nodes: $componentTemplateNodes } --><!-- /ko -->
</div>
