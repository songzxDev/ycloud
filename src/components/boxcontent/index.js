import template from './index.html'
import ko from 'knockout'
function init (params) {
  this.title = params.title
  this.vm = params.vm
  this.animated = params.animated || ko.observable(false)
  // 外界动态改变动画效果
  this.classes = ko.computed(() => {
    if (this.animated && this.animated.subscribe) {
      return this.animated() ? 'slideInUp animated' : ''
    } else {
      return this.animated ? 'slideInUp animated' : ''
    }
  })
}

export default {
  name: 'boxcontent',
  init,
  template
}
