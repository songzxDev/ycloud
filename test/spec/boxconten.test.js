import ko from 'knockout'
import boxcontent from '../../src/components/boxcontent/index.js'
import $ from 'jquery'
describe('y-box-content组件', () => {
  const PREFIX = 'y-'
  ko.components.register(PREFIX + boxcontent.name, {
    viewModel: boxcontent.init,
    template: boxcontent.template,
    synchronous: true
  })
  ko.cleanNode(document.getElementsByTagName('body')[0])
  let vm = {
    title: ko.observable('test')
  }
  document.body.innerHTML = '<y-boxcontent params="title: title"><span>content</span></y-boxcontent>'
  ko.applyBindings(vm)
  var el = $('body').find('.y-box-content')
  it('内部样式测试', () => {
    expect(el.hasClass('y-box-content')).toEqual(true)
    expect(el.find('.y-box-content-title').length > 0).toEqual(true)
  })
  it('属性title测试', () => {
    var text = el.find('.y-box-content-title')
    expect(text.text()).toEqual('test')
  })
  it('快照', () => {
    expect($('body').get(0)).toMatchSnapshot()
  })
})
