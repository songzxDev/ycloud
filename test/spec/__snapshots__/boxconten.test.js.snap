// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`y-box-content组件 快照 1`] = `
<body>
  <y-boxcontent
    params="title: title"
  >
    <div
      class="y-box-content"
      data-bind="css:classes,style:style"
    >
      
  
      <!-- ko if: title -->
      
  
      <div
        class="y-box-content-title"
        data-bind="text:title"
      >
        test
      </div>
      
  
      <!-- /ko -->
      
  
      <!-- ko template: { nodes: $componentTemplateNodes } -->
      <span>
        content
      </span>
      <!-- /ko -->
      

    </div>
    

  </y-boxcontent>
</body>
`;
